### ISSUE TITLE
[MCS-DE-GO] Syntax Validation does not catch all .LU syntax errors

### ISSUE THREAD

**Issue by goergenj** [microsoft/botframework-cli#1092]
On our implementation we found that not all .LU-file syntax errors are being recognized.

We had this (erroneous) entry:

#GetConfirmation
- yes of course
- I'd like that yes
- I'd love to
- no thank you
- not today thanks
- no rather not

```
# GetConfirmation
- {confirmation = yes} of course
- I'd like that {confirmation = yes}

@ list confirmation
	- yes :
	     - of course; sure;
    - no :
	     - no way; i don't
```

This is the correct syntax:

```
# GetConfirmation
- yes of course
- I'd like that yes

@ list confirmation =
	- yes :
	     - of course; sure
    - no :
	     - no way; i don't
```

Errors were:
- Labelled utterances on an entity list without roles
- Missing "=" in list entity definition
- ";" at the end of synonym list

Please check, if the .LU-file Syntaxchecker can be updated/improved.

@joaollq @sgellock @benbrown

**Comment by feich-ms** [comment]
@cwhitten thanks. I transferred this from composer to botframework-cli repo and I will take the fix in R12 developing cycle of bf-cli

**Comment by feich-ms** [comment]
@goergenj thanks for reporting these valuable issues. After figured out these cases, we took the solutions as below.

1. Labelled utterances on an entity list without roles
    **feich-ms: Already fixed in lasted release. Current GA composer(v1.3.0) can throw validation errors when you are editing invalid lu content as below screenshot.**
![image](https://user-images.githubusercontent.com/43032123/104896940-c17a4a80-59b2-11eb-8609-8014753d66f3.png)

2. Missing "=" in list entity definition
    **feich-ms: "=" is designed optional in our lu parser, so missing ''=" will not cause sematic parsing fail, it has no negative impacts on parsing result, so we will not add strict validation for "=" missing here.**

3. ";" at the end of synonym list
   **feich-ms: yes, this is a bug that can cause parsing failures or unexpected results, we just made a fix here #1093 to resolve this bug. It will take effect at next bf-cli release targeting the end of next month**
